### ISSUE TITLE
Undocumented `Missing "driver" option` error in 9.2.2

### ISSUE THREAD

**Issue by brandon-leapyear** [nestjs/graphql#1979]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

`@nestjs/graphql` 9.2.2 throws an error
```
ERROR [GraphQLModule] Missing "driver" option. In the latest version of "@nestjs/graphql" package (v10) a new required configuration property called "driver" has been introduced. Check out the official documentation for more details on how to migrate ("Quick start" chapter).
```
Which is a bit odd, for a number of reasons:
1. It's not on v10 yet
2. I can find no mention of this on the ["Quick start" page](https://docs.nestjs.com/graphql/quick-start)
3. The `CHANGELOG` in this repo has no mention of this
    1. Related: the CHANGELOG also only goes up to 9.2.0, but npmjs.org shows 9.2.1, 9.2.2, and 9.2.5??

### Expected behavior

(It should not error OR the official docs should be updated) AND changelogs should be updated

It seems like this feature was released too early?

### Package version

9.2.5

### Graphql version

`graphql`: 
`apollo-server-express`: 
`apollo-server-fastify`: 

### In which operating systems have you tested?

- [ ] macOS
- [ ] Windows
- [ ] Linux

**Comment by brandon-leapyear** [comment]
Is 9.2 just broken? Installing 9.2.0 in our code causes:
```
    Nest can't resolve dependencies of the GraphQLModule (HttpAdapterHost, GqlModuleOptions, ?, GraphQLTypesLoader). Please make sure that the argument AbstractGraphQLDriver at index [2] is available in the GraphQLModule context.

    Potential solutions:
    - If AbstractGraphQLDriver is a provider, is it part of the current GraphQLModule?
    - If AbstractGraphQLDriver is exported from a separate @Module, is that module imported within GraphQLModule?
      @Module({
        imports: [ /* the Module containing AbstractGraphQLDriver */ ]
      })
```

Trying to run the [sample](https://github.com/nestjs/nest/tree/master/sample/23-graphql-code-first), which works on 9.1.2, but then using 9.2.0 in the sample causes
```
 Error  The "@nestjs/graphql" plugin is not compatible with Nest CLI. Neither "after()" nor "before()" nor "afterDeclarations()" function have been provided.
```

**Comment by kamilmysliwiec** [comment]
The latest published version is 9.1.2 https://www.npmjs.com/package/@nestjs/graphql

It seems that you've installed a version published under the `next` tag (https://www.npmjs.com/package/@nestjs/graphql see Versions tab), not `latest`.

**Comment by brandon-leapyear** [comment]
Odd. When I change my `package.json`:
```diff
-"@nestjs/graphql": "< 9.2.0"
+"@nestjs/graphql": "~9.1.2"
```
Yarn installs 9.2.7. Is that a Yarn bug?
